group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/PartitionWriter.java

```diff
@@ -18,40 +18,21 @@
 
 import org.apache.kafka.common.KafkaException;
 import org.apache.kafka.common.TopicPartition;
-import org.apache.kafka.common.requests.TransactionResult;
+import org.apache.kafka.common.record.MemoryRecords;
+import org.apache.kafka.storage.internals.log.LogConfig;
 import org.apache.kafka.storage.internals.log.VerificationGuard;
 
-import java.util.List;
 import java.util.concurrent.CompletableFuture;
 
 /**
  * A simple interface to write records to Partitions/Logs. It contains the minimum
  * required for coordinators.
- *
- * @param <T> The record type.
  */
-public interface PartitionWriter<T> {
-
-    /**
-     * Serializer to translate T to bytes.
-     *
-     * @param <T> The record type.
-     */
-    interface Serializer<T> {
-        /**
-         * Serializes the key of the record.
-         */
-        byte[] serializeKey(T record);
-
-        /**
-         * Serializes the value of the record.
-         */
-        byte[] serializeValue(T record);
-    }
+public interface PartitionWriter {
 
     /**
      * Listener allowing to listen to high watermark changes. This is meant
-     * to be used in conjunction with {{@link PartitionWriter#append(TopicPartition, List)}}.
+     * to be used in conjunction with {{@link PartitionWriter#append(TopicPartition, VerificationGuard, MemoryRecords)}}.
```

Comment:
Is there a programatic way to check if these links are broken due to refactoring, or do you need to do it manually?

Just wondering if there is an easy way to check you did them all :)

Reply:
Intellij reports them as warning. I suppose that we would get warning when we generate the javadoc too.